pennylane/ops/op_math/parametrized_hamiltonian.py

```diff
@@ -53,45 +53,43 @@ class ParametrizedHamiltonian:
 
     """
 
-    num_wires = qml.operation.AnyWires
-    grad_method = "A"  # supports analytic gradients
-
-    def __init__(
-        self,
-        coeffs,
-        observables,
-    ):
+    def __init__(self, coeffs, observables):
 
         if qml.math.shape(coeffs)[0] != len(observables):
             raise ValueError(
                 "Could not create valid Hamiltonian; "
                 "number of coefficients and operators does not match."
             )
 
-        for obs in observables:
-            if not isinstance(obs, Observable):
-                raise ValueError(
-                    "Could not create circuits. Some or all observables are not valid."
-                )
+        # ToDo: assuming this is how we want to do things, make this clear in the documentation
+        # since organizing into H_fixed and H_parametrized potentially moves around the order of the terms compared
+        # to how the terms were entered, sorting seems the least ambiguous organization for the wires
+        self.wires = Wires.all_wires([op.wires for op in observables], sort=True)
```

Comment:
I'd say not to sort the wires. Check the `Composite` operator for a similar example.

Reply:
I agree that implicitly sorting here might be surprising to users.
I'm not sure how exactly, but one complication could be that terms are affected by the sorting and applied to wrong wires, or that the terms change order as well and the implicit parameter mapping could be affected.

This is to say, I don't see a bug caused by this, but it appears a fragile construction because the order of terms matters overall.

Reply:
The difference here is that unlike in `Composite`, there is a difference between the order of the wires when the operators were given, and the order of the wires that the operator returned by `H(params, t)`, unless the user always puts the fixed terms before the functions. If any callables come before any fixed coefficients in coeffs, we reorder the terms to separate `H_fixed` and `H_parametrized`. Example:


```
>>> def f2(x, t): return x*t
>>> coeffs = [f2, 2, 3]

# note that wires are specified in order 2, 1, 0
>>> ops = [qml.PauliX(2), qml.PauliX(1), qml.PauliX(0)]  
>>> H = ParametrizedHamiltonian(coeffs, ops)

>>> H(2, 1).wires
<Wires = [1, 0, 2]>
```

The question is then what `H.wires` should return. It could match the order the wires were entered in (`[op.wires for op in observables]`), it could match the order the wires will be in on a call of `H(params, t)` (`[op.wires for op in H_ops_fixed] + [op.wires for op in H_ops_parametrized]`), or it can be sorted. No matter what, between the input order, `H.wires`, and `H(params, t).wires`, there will be some mismatch unless the user always specified `H_drift` terms first. 

But maybe the clearest solution is just to specify in the docs that they should include all fixed terms first, and then all parametrized terms after, and then we can use the input wire order throughout.

Reply:
Do we really need to have `H.H_fixed` and `H.H_parametrized`? If not we can remove the separation of callables and scalars in the `__init__`, removing:
```python
self.H_coeffs_fixed = []
self.H_coeffs_parametrized = []
self.H_ops_fixed = []
self.H_ops_parametrized = []

for coeff, obs in zip(coeffs, observables):
    if callable(coeff):
        self.H_coeffs_parametrized.append(coeff)
        self.H_ops_parametrized.append(obs)
    else:
        self.H_coeffs_fixed.append(coeff)
        self.H_ops_fixed.append(obs)
```

And then do the `if callable(coeff)` check inside the `__call__` method.

This way we can ensure that the order of `ops` is mantained.

Reply:
Good question. I like that it allows you to ask for either the constant or the parametrized term, since that tends to be form its written in, but I don't know that that's actually useful - and its not part of the UI description. And this would certainly be clearer, since it doesn't rearrange anything compared to the input.

Reply:
I quite like the possibility to obtain the constant and parametrized part. I am not sure I see why sorting the wires of the constructed composite and parametrized Hamiltonian could be seen as unintuitive 🤔 This is also how we do it currently in `qml.Hamiltonian` (not that the state of the art is ever a good pro argument, but I do agree with this convention)

```python
coeff = np.ones(5)
ops = [qml.PauliX(i) @ qml.PauliX((i-1)%5) for i in range(5,0,-1)]
H = qml.Hamiltonian(coeff, ops)
print(H, H.wires)
```

```pycon
  (1.0) [X5 X4]
+ (1.0) [X4 X3]
+ (1.0) [X3 X2]
+ (1.0) [X2 X1]
+ (1.0) [X1 X0] <Wires = [0, 1, 2, 3, 4, 5]>
```

Reply:
Update: the wires now have the same order as the output operator (i.e. first `H_fixed wires`, then `H_parameterized` operators). Assuming the user inputs all `H_fixed` terms before `H_parametrized` terms, this will make `H.wires`, `H(params, t).wires`, and the input order of the wires identical.